python/orca/tutorial/pytorch/NCF/train_xshards.py

```diff
@@ -0,0 +1,140 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+# Most of the pytorch code is adapted from guoyang9's NCF implementation for
+# ml-1m dataset.
+# guoyang9's source code: https://github.com/guoyang9/NCF
+#
+
+import os
+import time
+import numpy as np
+import pandas as pd 
+import scipy.sparse as sp
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import torch.utils.data as data
+import torch.nn.functional as F 
+from model import NCF
+
+#Step 1: Init Orca Context
+
+from bigdl.orca import init_orca_context, stop_orca_context
+init_orca_context(cores=1, memory="8g")
+
+#Step 2: Define Dataset
+
+from bigdl.orca.data import XShards
+from bigdl.orca.data.pandas import read_csv
+from sklearn.model_selection import train_test_split
+
+# prepare the train and test datasets
+user_num = 6040
+item_num = 3952
+
+def preprocess_data():   
+    data_X = read_csv(
+        "ml-1m/ratings.dat", 
+        sep="::", header=None, names=['user', 'item'], 
+        usecols=[0, 1], dtype={0: np.int32, 1: np.int32})  
+    data_X = data_X.partition_by("user", data_X.num_partitions())

```

Comment:
num_partition default will keep the original partition, so no need the second argument?